source/extensions/common/aws/signer_impl.h

```diff
@@ -45,10 +48,17 @@ using SignatureConstants = ConstSingleton<SignatureConstantValues>;
 class SignerImpl : public Signer, public Logger::Loggable<Logger::Id::http> {
 public:
   SignerImpl(absl::string_view service_name, absl::string_view region,
-             const CredentialsProviderSharedPtr& credentials_provider, TimeSource& time_source)
+             const CredentialsProviderSharedPtr& credentials_provider, TimeSource& time_source,
+             const std::vector<envoy::type::matcher::v3::StringMatcher>& matcher_config)
```

Comment:
I suggest changing this so that you consume the vector of matcher pointers from the config when the signer is instantiated.

If this doesn't make sense because of lambda, and gRPC, then "this is fine" :)